### ISSUE TITLE
[feature - radiators] Kerbal's body heat

### ISSUE THREAD

**Issue by OzoneGrif** [ShotgunNinja/Kerbalism#16]
This is a feature request to give some love to the game's radiators which are really useless right now. The game and Kerbalism handles heat decently, but it's missing one very important thing : Body heat.

When you have ten kerbals inside a space station, the joules generated by their biological bodies in this tiny confined space should make the ambiant air heat up very quickly and cook them alive without life support. I haven't noticed a difference with Life Support when you have one or ten kerbals onboard.

Then, with the Life Support on, it should transfert this biological heat to the parts themselves (internal heat). This heat will propagate through the vessel and radiate to space through the skin... but with many kerbals onboard, the skin just won't be enough, the parts will heat up, and Life Support should be overloaded (unable to cool down the air if the internal heat is too high).

Here comes radiators, cooling the parts, helping Life Support in its function, and saving the nice Kerbals.

To recap : 
- X kerbal onboard = X * internal part heating
- life support is inefficient if internal heat is too high

That's it.

This is a dream I've had for a looong time, Kerbalism is an amazing mod and I really love it, I would totally donate to see this feature implemented :)

**Comment by ShotgunNinja** [comment]
You have to consider that there are no parts when the vessel is unloaded. Therefore, to handler part heat in background (as opposed to environment temperature where the vessel will be) I'll have to re-implement the whole stock temperature system. Not going to happen. I too share the opinion that radiators are useless right now, and your idea is interesting. Its just technically unfeasible for unloaded vessels.

A simpler version, technically implementable for unloaded vessels, would be:
- new pseudo-resource, WasteHeat
- Kerbals produce waste heat (some other module can produce that too, eventually)
- radiators consume waste heat constantly (if active), even in background

Note that's missing the part->part heat tranfer completely, somehow replaced by the 'resource dynamics'.

**Comment by OzoneGrif** [comment]
Thanks for your answer. I'm curious because I remember something about Squad saying that heat is simulated in background since KSP 1.05 : http://wiki.kerbalspaceprogram.com/wiki/1.0.5
- Unloaded vessels will have thermo applied for the unloaded time on reload.

Is the new thermal moddable interface able to manipulate this ?

I am not too fond of adding a fake resource. I love mods when they integrate properly with the game, and this is why I love Kerbalism. Unless it was possible to hide the resource from the player and use it internally only (technical soup)... create the resource when switching to background mode, and generate instant heat and destroy the resource when loading the vessel.

**Comment by ShotgunNinja** [comment]
Yes, pseudo-resources are massless and invisible to the user. Its just an implementation detail.

**Comment by OzoneGrif** [comment]
Why not then, it should be close enough to make the internal soup invisible to the player.
Real internal heat when the vessel is active, fake global resource while in background.

**Comment by ShotgunNinja** [comment]
The problem I see is in what way this WasteHeat is going to make the situation worse for the Kerbals. Also in some way this seem to overlap with the climatization mechanic.

**Comment by ShotgunNinja** [comment]
Actually, with the new framework for arbitrary rules in 0.9.9.5 this could be implemented this way:

- create a 'negative waste heat' pseudo-resource
- create a rule that consume this negative waste heat per-kerbal
- the kerbals will die/breakdown (its configurable per rule) when the negative waste heat is gone
- add a ModuleGenerator to the radiators, that slowly produce 'negative waste heat'
